ProjectFiles/Assets/Scripts/Network/Server.cs

```diff
@@ -87,35 +94,78 @@ public void HandleNetworkEvents()
                         {
                             var readerContext = default(DataStreamReader.Context);
                             var ev = (ClientNetworkEvent) reader.ReadByte(ref readerContext);
-                            HandleEvent(connections[i], endpoint, ev, reader, readerContext);
+                            HandleEvent(connections[i], endpoint, ev, reader, readerContext, i);
```

Comment:
I think this won't always work? I can't currently test it, but the `i` variable may not always refer to the same connection since the connections list may change. Instead I think we should use the internal connection ID.

```suggestion
                            HandleEvent(connections[i], endpoint, ev, reader, readerContext, connections[i].InternalId);
```